1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,6 +1,46 @@
-function setAlarm() {}
+let isPaused = false;
+let timer;
+let mini;
+let seconds;
 
-// DO NOT EDIT BELOW HERE
+function setAlarm() {
+  document.querySelector("body").style.backgroundColor = "white";
+  let timeRemaining = document.getElementById("timeRemaining");
+  let alarmSet = document.getElementById("alarmSet").value * 60;
+
+  if (alarmSet > 0) {
+    timer = setInterval(() => {
+      if (!isPaused) {
+        alarmSet--;
+        mini = Math.floor(alarmSet / 60);
+        seconds = alarmSet % 60;
+        seconds = seconds < 10 ? `0${seconds}` : seconds;
+        mini = mini < 10 ? `0${mini}` : mini;
+        timeRemaining.textContent = `Time Remaining: ${mini}:${seconds}`;
+      }
+      if (mini == 0 && seconds == 0) {
+        playAlarm();
+        clearInterval(timer);
+        timeRemaining.textContent = `Time Remaining: 00:00`;
+        document.querySelector("body").style.backgroundColor = "red";
+        // timeRemaining.textContent = `Time Remaining: 00:00`;
+      }
+    }, 1000);
+  }
+  document.getElementById("alarmSet").value = "";
+}
```

Comment:
This function works well. You could also create a named function outside of setInterval() and pass it in - think about the pros and cons of each approach.